src/herder/HerderImpl.cpp

```diff
@@ -98,7 +100,44 @@ HerderImpl::~HerderImpl()
 Herder::State
 HerderImpl::getState() const
 {
-    return mHerderSCPDriver.getState();
+    return mState;
+}
+
+void
+HerderImpl::setTrackingSCPState(uint64_t index, StellarValue const& value,
+                                bool isTrackingNetwork)
+{
+    mTrackingSCP = ConsensusData{index, value.closeTime};
+    setState(isTrackingNetwork ? Herder::HERDER_TRACKING_NETWORK_STATE
+                               : Herder::HERDER_TRACKING_LCL_STATE);
+}
+
+uint32
+HerderImpl::trackingConsensusLedgerIndex() const
+{
+    releaseAssert(getState() != Herder::State::HERDER_BOOTING_STATE);
+    releaseAssert(mTrackingSCP.mConsensusIndex <= UINT32_MAX);
+    return static_cast<uint32>(mTrackingSCP.mConsensusIndex);
+}
+
+TimePoint
+HerderImpl::trackingConsensusCloseTime() const
+{
+    releaseAssert(getState() != Herder::State::HERDER_BOOTING_STATE);
+    return mTrackingSCP.mConsensusCloseTime;
+}
+
+void
+HerderImpl::setState(State st)
+{
+    mState = st;
```

Comment:
can we enforce that we don't go back to booting state? (
and related to this, I think it's invalid to go back to `HERDER_TRACKING_LCL_STATE`)